students/JerryH/Lesson09/mailroom.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+class Donor:
+
+    def __init__(self, first, last, donations = None):
+        self.first = first
+        self.last = last
+        self.donations = donations
+
+    @property
+    def full_name(self):
+        return "{} {}".format(self.first, self.last)
+
+    @property
+    def donor_history(self):
+        return sum(self.donations)
+
+    def add_donation(self, amount):
+        return self.donations.append(amount)
+
+
+class DonorBook:
+
+    def __init__(self, donors = []):
+        if donors is None:
+            self.donors = []
+        else:
+            self.donors = donors
+
+    def add_donor(self, donor):
+        # if donor not in self.donors:
+        self.donors.append(donor)
+
+    def get_all_donor_names(self):
+        donor_name_list = []
+        for donor in self.donors:
+            #print("-->" + donor.full_name)
+            donor_name_list.append(donor.full_name)
+        return donor_name_list
```

Comment:
great place to use comprehension!
`return [donor.full_name for donor in self.donors]`